src/JsonApi/Serializer/CollectionNormalizer.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\JsonApi\Serializer;
+
+use ApiPlatform\Core\Api\ResourceClassResolverInterface;
+use ApiPlatform\Core\DataProvider\PaginatorInterface;
+use ApiPlatform\Core\Exception\RuntimeException;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Serializer\ContextTrait;
+use ApiPlatform\Core\Util\IriHelper;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Normalizes collections in the JSON API format.
+ *
+ * @author Kevin Dunglas <dunglas@gmail.com>
+ * @author Hamza Amrouche <hamza@les-tilleuls.coop>
+ */
+final class CollectionNormalizer implements NormalizerInterface, NormalizerAwareInterface
+{
+    use ContextTrait;
+    use NormalizerAwareTrait;
+
+    const FORMAT = 'jsonapi';
+
+    private $resourceClassResolver;
+    private $pageParameterName;
+    private $resourceMetadataFactory;
+    private $propertyMetadataFactory;
+
+    public function __construct(ResourceClassResolverInterface $resourceClassResolver, ResourceMetadataFactoryInterface $resourceMetadataFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, string $pageParameterName)
+    {
+        $this->resourceClassResolver = $resourceClassResolver;
+        $this->pageParameterName = $pageParameterName;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->propertyMetadataFactory = $propertyMetadataFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsNormalization($data, $format = null)
+    {
+        return self::FORMAT === $format && (is_array($data) || ($data instanceof \Traversable));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($data, $format = null, array $context = [])
+    {
+        $currentPage = $lastPage = $itemsPerPage = 1.;
+
+        // If we are normalizing stuff one level down (i.e., an attribute which could be already an array)
+        $returnDataArray = [];
+        if (isset($context['api_sub_level'])) {
+            foreach ($data as $index => $obj) {
+                $returnDataArray['data'][][$index] = $this->normalizer->normalize($obj, $format, $context);
```

Comment:
Looks correct according to what are the other normalizer's doing in case of a `api_sub_level` flag.
https://github.com/api-platform/core/blob/b88d975fa152d295ffeed332dab81dd8376722d1/src/Hal/Serializer/CollectionNormalizer.php#L60
https://github.com/api-platform/core/blob/183b0a58114a3294380c27c6ff7b2536b128523a/src/Hydra/Serializer/CollectionNormalizer.php#L65